build_tools/cmake/iree_mlir_benchmark_suite.cmake

```diff
@@ -152,64 +153,82 @@ function(iree_mlir_benchmark_suite)
     foreach (_BENCHMARK_MODE IN LISTS _RULE_BENCHMARK_MODES)
       set(_BENCHMARK_DIR_NAME
           "iree-${_RULE_DRIVER}__${_RULE_TARGET_ARCHITECTURE}__${_BENCHMARK_MODE}")
-      set(_ARTIFACTS_DIR "${_ROOT_ARTIFACTS_DIR}/${_MODULE_DIR_NAME}/${_BENCHMARK_DIR_NAME}")
-
-      set(_TRANSLATION_ARGS "--iree-mlir-to-vm-bytecode-module")
-      list(APPEND _TRANSLATION_ARGS "--iree-hal-target-backends=${_RULE_TARGET_BACKEND}")
-      list(APPEND _TRANSLATION_ARGS ${_RULE_TRANSLATION_FLAGS})
-
-      set(_VMFB_FILE "${_ARTIFACTS_DIR}/compiled.vmfb")
-      add_custom_command(
-        OUTPUT "${_VMFB_FILE}"
-        COMMAND
-          "$<TARGET_FILE:iree_tools_iree-translate>"
-            ${_TRANSLATION_ARGS}
-            "${_SOURCE_FILE}"
-            -o "${_VMFB_FILE}"
-        WORKING_DIRECTORY "${_ARTIFACTS_DIR}"
-        DEPENDS
-          iree_tools_iree-translate
-          "${_DOWNLOAD_TARGET_NAME}"
-        COMMENT "Generating ${_VMFB_FILE}"
-      )
 
+      # A list of name segments for composing unique CMake target names.
       set(_COMMON_NAME_SEGMENTS "${_MODULE_NAME}")
       string(REPLACE "," "-" _TAGS "${_MODULE_TAGS}")
       string(REPLACE "," "-" _MODE "${_BENCHMARK_MODE}")
       list(APPEND _COMMON_NAME_SEGMENTS
            "${_TAGS}" "${_MODE}" "${_RULE_TARGET_BACKEND}"
            "${_RULE_TARGET_ARCHITECTURE}")
 
-      # Construct the benchmark artifact generation target name, which is the module
-      # name, followed by benchmark mode, target backend, and configuration.
-      set(_TRANSLATION_TARGET_NAME_LIST "iree-generate-benchmark-artifact")
-      list(APPEND _TRANSLATION_TARGET_NAME_LIST ${_COMMON_NAME_SEGMENTS})
-      list(JOIN _TRANSLATION_TARGET_NAME_LIST "__" _TRANSLATION_TARGET_NAME)
+      # The full list of translation flags.
+      set(_TRANSLATION_ARGS "--iree-mlir-to-vm-bytecode-module")
+      if("${_MODULE_SOURCE}" STREQUAL "TensorFlow")
+        list(APPEND _TRANSLATION_ARGS "--iree-input-type=mhlo")
```

Comment:
We should keep this flag back where it was - as we do other input types (tosa, just linalg, mhlo but from a .mlir instead of a protobuf, etc) we'll want to change that and having it here hidden behind another more generic flag makes that really hard to see.

Reply:
In fact I'm not sure I see the benefit of having MODULE_SOURCE at all - it just seems to be spooky action at a distance as in the cmake file adding a new test requires **7 different cmake variables** with precise ordering match. We should try to cut that down instead of adding more logic derived from it - for example, the source, entry function, and inputs could be expressed as a single cmake list of "source;function;input;input;..." - that makes it much easier to see what's being run and adding a new model becomes a single line.

Reply:
Yeah it's a bit unfortunate that we have so many variables to handle. It's partially due to current status and partially due to future extensibility. 

Note that really we should generate model benchmark artifacts from the original framework source so we don't even need to specify those source, entry function, function inputs, etc. because they are readily in the original source where we can just deduce. We don't have that yet and we are downloading from the cloud storage so it's forced to have so many fields to describe. 

For future extensibility side, having such fields makes us able to differentiate between the same models generated from different sources (e.g., TF, TFLite, PyTorch, etc.) and we can have a consistent string to describe them no matter how we change stuff in implementation (because the string is load-bearing in identifying benchmarks), e.g., I'd assume these existing model benchmarks to go to Python layer eventually and we don't need to change their benchmark name strings.

But yeah, `iree_mlir_benchmark_suite` itself probably should just be taking normal MLIR inputs and don't care about the framework source, etc. We can trim down the number of parameters. It depends on further future cleanup and we can go with that after they are ready. :)